src/hlsBinaries.ts

```diff
@@ -354,20 +354,21 @@ export async function downloadHaskellLanguageServer(
   // When searching for binaries, use startsWith because the compression may differ
   // between .zip and .gz
   const assetName = `haskell-language-server-${githubOS}-${ghcVersion}${exeExt}`;
-  logger.info(`Search for binary ${assetName} in release assests`);
+  logger.info(`Search for binary ${assetName} in release assets`);
+  const release = releases?.find(r => r.assets.find((x) => x.name.startsWith(assetName)));
   const asset = release?.assets.find((x) => x.name.startsWith(assetName));
   if (!asset) {
     logger.error(
-      `No binary ${assetName} found in the release assets: ${release?.assets.map((value) => value.name).join(',')}`
+      `No binary ${assetName} found in the release assets`
     );
-    window.showInformationMessage(new NoBinariesError(release.tag_name, ghcVersion).message);
+    window.showInformationMessage(new NoBinariesError(releases[0].tag_name, ghcVersion).message);
```

Comment:
release[0] is not necessarily correct for older ghc versions.

Reply:
what is releases[0]? the last one (1.5.0) or the first one (0.5.0?)?

Reply:
It's the last one, see: https://api.github.com/repos/haskell/haskell-language-server/releases